### PR TITLE
[fix](array-type) fix the be core dump when use collect_list result t…

### PR DESCRIPTION
**Pull request by carlvinhust2012**
# Proposed changes
1. this pr is used to fix the be core dump when use collect_list result to insert.
2. the core dump stack is as below:
*** Current BE git commitID: 55fdb555b ***
*** SIGSEGV unkown detail explain (@0x0) received by PID 63892 (TID 0x7f1004d7c700) from PID 0; stack trace: ***
 0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /home/disk1/hugo_work/doris_dev/baidu/bdg/doris/core/be/src/common/signal_handler.h:420
 1# 0x00007F10583D4920 in /lib64/libc.so.6
 2# __dynamic_cast in /home/disk1/hugo_work/doris_env/output/be/lib/palo_be
 3# doris::vectorized::ColumnVector<unsigned char>::insert_range_from(doris::vectorized::IColumn const&, unsigned long, unsigned long) at /home/disk1/hugo_work/doris_dev/baidu/bdg/doris/core/be/src/vec/columns/column_vector.cpp:255
 4# doris::vectorized::ColumnNullable::insert_range_from(doris::vectorized::IColumn const&, unsigned long, unsigned long) at /home/disk1/hugo_work/doris_dev/baidu/bdg/doris/core/be/src/vec/columns/column_nullable.cpp:179
 5# doris::vectorized::ColumnArray::insert_from(doris::vectorized::IColumn const&, unsigned long) at /home/disk1/hugo_work/doris_dev/baidu/bdg/doris/core/be/src/vec/columns/column_array.cpp:239
 6# doris::vectorized::ColumnNullable::insert_from(doris::vectorized::IColumn const&, unsigned long) at /home/disk1/hugo_work/doris_dev/baidu/bdg/doris/core/be/src/vec/columns/column_nullable.cpp:204
 7# doris::vectorized::MutableBlock::add_row(doris::vectorized::Block const*, int) at /home/disk1/hugo_work/doris_dev/baidu/bdg/doris/core/be/src/vec/core/block.cpp:966
 8# doris::stream_load::VNodeChannel::add_row(std::pair<doris::vectorized::Block*, int> const&, long) at /home/disk1/hugo_work/doris_dev/baidu/bdg/doris/core/be/src/vec/sink/vtablet_sink.cpp:188
 9# void doris::stream_load::IndexChannel::add_row<std::pair<doris::vectorized::Block*, int> >(std::pair<doris::vectorized::Block*, int> const&, long) at /home/disk1/hugo_work/doris_dev/baidu/bdg/doris/core/be/src/exec/tablet_sink.h:403
10# doris::stream_load::VOlapTableSink::send(doris::RuntimeState*, doris::vectorized::Block*) at /home/disk1/hugo_work/doris_dev/baidu/bdg/doris/core/be/src/vec/sink/vtablet_sink.cpp:522
11# doris::PlanFragmentExecutor::open_vectorized_internal() at /home/disk1/hugo_work/doris_dev/baidu/bdg/doris/core/be/src/runtime/plan_fragment_executor.cpp:308
12# doris::PlanFragmentExecutor::open() at /home/disk1/hugo_work/doris_dev/baidu/bdg/doris/core/be/src/runtime/plan_fragment_executor.cpp:249
13# doris::FragmentExecState::execute() at /home/disk1/hugo_work/doris_dev/baidu/bdg/doris/core/be/src/runtime/fragment_mgr.cpp:246
14# doris::FragmentMgr::_exec_actual(std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::PlanFragmentExecutor*)>) at /home/disk1/hugo_work/doris_dev/baidu/bdg/doris/core/be/src/runtime/fragment_mgr.cpp:500
15# std::_Function_handler<void (), doris::FragmentMgr::exec_plan_fragment(doris::TExecPlanFragmentParams const&, std::function<void (doris::PlanFragmentExecutor*)>)::{lambda()}>::_M_invoke(std::_Any_data const&) at /home/disk1/hugo_work/doris_dev/baidu/bdg/doris/palo-toolchain/ldb_toolchain/include/c++/11/bits/std_function.h:291
16# doris::ThreadPool::dispatch_thread() at /home/disk1/hugo_work/doris_dev/baidu/bdg/doris/core/be/src/util/threadpool.cpp:548
17# doris::Thread::supervise_thread(void*) at /home/disk1/hugo_work/doris_dev/baidu/bdg/doris/core/be/src/util/thread.cpp:426
18# start_thread in /lib64/libpthread.so.0
19# clone in /lib64/libc.so.6

3.after the fix, we can run the insert statement successfully.
MySQL [(none)]> insert into regression_test_load_p0_broker_load.tbl_test_array_fun select k1, collect_list(k2),  collect_list(k3), collect_list(k4), collect_list(k5), collect_list(k6),
    -> collect_list(k7), collect_list(k8), collect_list(k9), collect_list(k10), collect_list(k11), [true], collect_list(k13) from
    -> regression_test_load_p0_broker_load.tbl_temp_array_fun group by k1;
Query OK, 1 row affected (0.764 sec)
{'label':'insert_236f81d7285a436d_941257b724cbcf7b', 'status':'VISIBLE', 'txnId':'4562'}

MySQL [(none)]> select * from regression_test_load_p0_broker_load.tbl_test_array_fun;
+------+------+---------+---------+--------------------------+-----------+--------------+-----------------------+--------+-------------+--------+------+----------------+
| k1   | k2   | k3      | k4      | k5                       | k6        | k7           | k8                    | k9     | k10         | k11    | k12  | k13            |
+------+------+---------+---------+--------------------------+-----------+--------------+-----------------------+--------+-------------+--------+------+----------------+
|  200 | [2]  | [32768] | [65535] | ['b                   '] | ['hello'] | [2022-07-13] | [2022-07-13 12:30:00] | [0.67] | [3.1415926] | [6.67] | [1]  | ['happy life'] |
+------+------+---------+---------+--------------------------+-----------+--------------+-----------------------+--------+-------------+--------+------+----------------+
1 row in set (0.011 sec)

Issue Number: #7570
## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
4. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
5. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
6. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
7. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/columns/column_array.cpp [hunk 1]
```diff
@@ -235,7 +235,13 @@ void ColumnArray::insert_from(const IColumn& src_, size_t n) {
     size_t size = src.size_at(n);
     size_t offset = src.offset_at(n);
 
-    get_data().insert_range_from(src.get_data(), offset, size);
+    // Note: here we should process the case of 'Nullable(Array(NotNullable(nest)))'
```
Comment (cambyzju) [inline be/src/vec/columns/column_array.cpp:238]:
```suggestion
    // Note: here we should process the case of 'Array(NotNullable(nest))'
```

File: be/src/vec/columns/column_array.cpp [hunk 2]
```diff
@@ -235,7 +235,13 @@ void ColumnArray::insert_from(const IColumn& src_, size_t n) {
...
+    if (get_data().is_nullable() && !src.get_data().is_nullable()) {
+        reinterpret_cast<ColumnNullable*>(&get_data())
+                ->insert_range_from_not_nullable(src.get_data(), offset, size);
+    } else {
```
Comment (cambyzju) [inline be/src/vec/columns/column_array.cpp:242]: add DCHEC for `!get_data().is_nullable() && src.get_data().is_nullable()` ?

Comment (carlvinhust2012) [reply be/src/vec/columns/column_array.cpp:242]: if add DCHEC to replace, the insert_from() will just support the only one case. It is unreasonable.

File: be/src/vec/columns/column_array.cpp (hunk 1 above)
Comment (carlvinhust2012) [reply be/src/vec/columns/column_array.cpp:238]: ok

File: be/src/vec/columns/column_array.cpp (hunk 2 above)
Comment (carlvinhust2012) [reply be/src/vec/columns/column_array.cpp:243]: done

**Review by cambyzju — APPROVED** [review]
LGTM

**Pull request merged**